api/src/test/java/org/apache/iceberg/expressions/TestAggregateEvaluator.java

```diff
@@ -0,0 +1,187 @@
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
+import org.apache.iceberg.types.Types.IntegerType;
+import org.apache.iceberg.types.Types.StringType;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class TestAggregateEvaluator {
+  private static final Schema SCHEMA =
+      new Schema(
+          required(1, "id", IntegerType.get()),
+          optional(2, "no_stats", IntegerType.get()),
+          optional(3, "all_nulls", StringType.get()),
+          optional(4, "some_nulls", StringType.get()));
+
+  private static final DataFile FILE =
+      new TestDataFile(
+          "file.avro",
+          Row.of(),
+          50,
+          // any value counts, including nulls
+          ImmutableMap.of(1, 50L, 3, 50L, 4, 50L),
+          // null value counts
+          ImmutableMap.of(1, 10L, 3, 50L, 4, 10L),
+          // nan value counts
+          null,
+          // lower bounds
+          ImmutableMap.of(1, toByteBuffer(IntegerType.get(), 33)),
+          // upper bounds
+          ImmutableMap.of(1, toByteBuffer(IntegerType.get(), 2345)));
+
+  private static final DataFile FILE_2 =
+      new TestDataFile(
+          "file_2.avro",
+          Row.of(),
+          20,
+          // any value counts, including nulls
+          ImmutableMap.of(1, 20L, 3, 20L),
+          // null value counts
+          ImmutableMap.of(1, 0L, 3, 20L),
+          // nan value counts
+          null,
+          // lower bounds
+          ImmutableMap.of(1, toByteBuffer(IntegerType.get(), 33)),
+          // upper bounds
+          ImmutableMap.of(1, toByteBuffer(IntegerType.get(), 100)));
+
+  private static final DataFile FILE_3 =
+      new TestDataFile(
+          "file_3.avro",
+          Row.of(),
+          20,
+          // any value counts, including nulls
+          ImmutableMap.of(1, 20L, 3, 20L),
+          // null value counts
+          ImmutableMap.of(1, 20L, 3, 20L),
+          // nan value counts
+          null,
+          // lower bounds
+          ImmutableMap.of(1, toByteBuffer(IntegerType.get(), -33)),
+          // upper bounds
+          ImmutableMap.of(1, toByteBuffer(IntegerType.get(), 3333)));
+
+  private static final DataFile FILE_4 =
+      new TestDataFile(
+          "file_4.avro",
+          Row.of(),
+          20,
+          // any value counts, including nulls
+          ImmutableMap.of(1, 20L, 3, 20L),
+          // null value counts
+          ImmutableMap.of(1, 10L, 3, 20L),
+          // nan value counts
+          null,
+          // lower bounds
+          ImmutableMap.of(1, toByteBuffer(IntegerType.get(), -1000)),
+          // upper bounds
+          ImmutableMap.of(1, toByteBuffer(IntegerType.get(), 1000)));
+
+  private static final DataFile[] dataFiles = {FILE, FILE_2, FILE_3, FILE_4};
+
+  @Test
+  public void testIntAggregate() {
+    List<Expression> list =
+        ImmutableList.of(Expressions.count("id"), Expressions.max("id"), Expressions.min("id"));
```

Comment:
Why not add count(*) here as well?

Reply:
In fact, I think it would make sense for all of the tests to support count(*)

Reply:
Added. Thanks